src/main/java/cn/nukkit/level/GlobalBlockPalette.java

```
@@ -555,6 +561,14 @@ public static int getOrCreateRuntimeId(int protocol, int legacyId) throws NoSuch
         }
     }
 
+    public static int getLegacyFullId(int protocolId, int runtimeId) {
+        if (protocolId < ProtocolInfo.v1_17_0) {
+            return runtimeIdToLegacy428.get(runtimeId);
+        } else {
+            return runtimeIdToLegacy440.get(runtimeId);
+        }
```

This should probably throw an exception if used with an unsupported protocol